src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClassInnerAndNestedClass.java

```diff
@@ -0,0 +1,76 @@
+/*
+FinalClass
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.design.finalclass;
+
+public class InputFinalClassInnerAndNestedClass {
+
+    private class SuperClass { // ok
+        private SuperClass() {
+        }
+    }
+
+    private class SubClass extends SuperClass {
+    }
+
+    class SameName { // violation
+        private SameName() {
+        }
+    }
+
+    static class Test {
+        static class SameName { // ok
+            private SameName() {
+            }
+            class Test3 {
```

Comment:
Maybe I am missing something from the requirements, but why are completely empty classes not checked?

Reply:
Good point. From the check [description](https://checkstyle.org/config_design.html#FinalClass_Description)
> Checks that a class which has only private constructors is declared as final.

And the [specs](https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.8.9) says that
> The default constructor has the same access modifier as the class.

Thus, if a **private** class has no ctor, it should be `final`.

Reply:
Started new issue #11338